lib/java/PlagiarismDetection/src/Tokenizer/ANTLRDynamicTokenizer.java

```diff
@@ -62,51 +62,57 @@ public static boolean understandsLanguage(String language) {
   /*
    * Creates or returns tokenizer from hashtable
    */
-	public static Tokenizer getTokenizer(String language) throws Exception {
+  public static Tokenizer getTokenizer(String language) throws Exception {
     if (ANTLRDynamicTokenizer.grammarsBinDir == null) {
       throw new Exception("getTokenizer() called without grammarsBinDir defined");
     }
     if (!ANTLRDynamicTokenizer.tokenizers.containsKey(language)) {
       ANTLRDynamicTokenizer.tokenizers.put(language, new ANTLRDynamicTokenizer(language));
     }
     return ANTLRDynamicTokenizer.tokenizers.get(language);
-	}
+  }
 
-	public void TokenizeSubmissions(ArrayList<Submission> submissions) {
-		for (Submission s : submissions) {
-			tokenize(s);
-		}
-	}
+  public void TokenizeSubmissions(ArrayList<Submission> submissions) {
+    Iterator<Submission> itr = submissions.iterator();
+    while (itr.hasNext()) {
+      Submission s = itr.next();
+      tokenize(s);
+      if (s.getCodeTokens().size() == 0) {
+        System.out.println("Empty file detected & would be removed from Plagiarism Detection");
```

Comment:
Better as "Empty file detected.  This file will be removed from the plagiarism detection procedure."

"Empty file detected & would be removed from Plagiarism Detection"

Error messages should go to stderr. 

System.err.  Maybe better to push this to a separate issue since it is system-wide.